### PR TITLE
Implement news panels for future use in dashboard overlay

### PR DESCRIPTION
**Pull request by EVAST9919**
Screenshot_1

### CODE REVIEW

File: osu.Game/Overlays/Dashboard/Home/News/HomeNewsPanelFooter.cs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Shapes;
+using osu.Game.Online.API.Requests.Responses;
+
+namespace osu.Game.Overlays.Dashboard.Home.News
+{
+    public abstract class HomeNewsPanelFooter : CompositeDrawable
```
Comment (bdach) [inline osu.Game/Overlays/Dashboard/Home/News/HomeNewsPanelFooter.cs:13]: I'm not sure I agree with the design of this. I'd actually slightly prefer if the two inheritors of this were separated.

The reasoning is that this control is too nebulous to be useful, and the differences in design don't help (having the misspelled `BarPading` to control how the separator is displayed feels all sorts of wrong).

File: osu.Game/Overlays/Dashboard/Home/News/HomeNewsPanel.cs [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                    Origin = Anchor.TopRight,
+                    Children = new Drawable[]
+                    {
+                        new OsuSpriteText
+                        {
+                            Anchor = Anchor.TopRight,
+                            Origin = Anchor.TopRight,
+                            Font = OsuFont.GetFont(weight: FontWeight.Bold), // using Bold since there is no 800 weight alternative
+                            Colour = colourProvider.Light1,
+                            Text = $"{date: dd}"
```
Comment (bdach) [inline osu.Game/Overlays/Dashboard/Home/News/HomeNewsPanel.cs:140]: nit: unnecessary space

```suggestion
                            Text = $"{date:dd}"
```

File: osu.Game/Overlays/Dashboard/Home/News/NewsPostDrawableDate.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Cursor;
+using osu.Game.Graphics;
+using osu.Framework.Graphics;
+
+namespace osu.Game.Overlays.Dashboard.Home.News
+{
+    public abstract class NewsPostDrawableDate : CompositeDrawable, IHasCustomTooltip
```
Comment (bdach) [inline osu.Game/Overlays/Dashboard/Home/News/NewsPostDrawableDate.cs:13]: This I definitely disagree with existing. This base class does nothing useful.

IMO this should be removed, inheritors should implement `IHasCustomTooltip` themselves. Not too many lines of overhead.

File: osu.Game/Overlays/Dashboard/Home/News/NewsTitleLink.cs [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+            this.post = post;
+
+            RelativeSizeAxes = Axes.X;
+            AutoSizeAxes = Axes.Y;
+        }
+
+        [BackgroundDependencyLoader]
+        private void load(GameHost host)
+        {
+            Child = new TextFlowContainer(t =>
```
Comment (bdach) [inline osu.Game/Overlays/Dashboard/Home/News/NewsTitleLink.cs:29]: I know we don't do underline text anywhere but the current yellow colour on hover definitely feels wrong. Would use something themed instead if possible.

Comment (EVAST9919) [reply osu.Game/Overlays/Dashboard/Home/News/NewsTitleLink.cs:29]: Would be nice to have any suggestions

Comment (EVAST9919) [reply osu.Game/Overlays/Dashboard/Home/News/NewsTitleLink.cs:29]: Alright, I just changed hover colour
Screenshot_1

File: osu.Game/Overlays/Dashboard/Home/HomePanel.cs [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Effects;
+using osu.Framework.Graphics.Shapes;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.Dashboard.Home
+{
+    public class HomePanel : Container
```
Comment (bdach) [inline osu.Game/Overlays/Dashboard/Home/HomePanel.cs:15]: perhaps should be `abstract` as I don't think anything should be instantiating this outright

File: osu.Game/Overlays/Dashboard/Home/News/HomeNewsGroupPanel.cs [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Game.Online.API.Requests.Responses;
+
+namespace osu.Game.Overlays.Dashboard.Home.News
+{
+    public class HomeNewsGroupPanel : HomePanel
```
Comment (bdach) [inline osu.Game/Overlays/Dashboard/Home/News/HomeNewsGroupPanel.cs:13]: maybe `NewsItemGroupPanel`

File: osu.Game/Overlays/Dashboard/Home/News/HomeShowMoreNewsPanel.cs [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+using System.Collections.Generic;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays.Dashboard.Home.News
+{
+    public class HomeShowMoreNewsPanel : OsuHoverContainer
```
Comment (bdach) [inline osu.Game/Overlays/Dashboard/Home/News/HomeShowMoreNewsPanel.cs:13]: `ShowMoreNewsPanel` (the `Home` prefix is obvious - see namespace)

**Review by bdach — COMMENTED** [review]
would do some renames here as prefixing everything with `Home` feels gratuitous

File: osu.Game/Overlays/Dashboard/Home/HomePanel.cs (hunk 5 above)
Comment (EVAST9919) [reply osu.Game/Overlays/Dashboard/Home/HomePanel.cs:15]: I'm using this in `HomeShowMoreNewsPanel` and will be used as a container for beatmaps later

**Review by bdach — APPROVED** [review]
seems inoffensive enough to me

**Review by peppy — APPROVED** [review]

**Pull request merged**